### PR TITLE
fix: Bypass the response body processing in ai-proxy if it is returned internally

### LINKED ISSUE #1145: ai-proxy  + request block  断开连接
### Ⅰ. Issue Description

ai-proxy  + request block 客户端请求触发黑名单时直接 断开连接

### Ⅱ. Describe what happened

请求返回：
```
curl -vvv http://172.20.5.14:20080/api/openai/v1/chat/completions -d '{"messages": [{"role": "user", "content": "helloworld"}], "model": "gpt-3.5"}' -H "content-type: application/json"
* Uses proxy env variable no_proxy == 'localhost,127.0.0.0/8,::1'
*   Trying 172.20.5.14:20080...
* Connected to 172.20.5.14 (172.20.5.14) port 20080
> POST /api/openai/v1/chat/completions HTTP/1.1
> Host: 172.20.5.14:20080
> User-Agent: curl/8.5.0
> Accept: */*
> content-type: application/json
> Content-Length: 77
> 
< HTTP/1.1 200 OK
< content-type: text/plain
< date: Tue, 23 Jul 2024 07:30:34 GMT
< server: istio-envoy
< transfer-encoding: chunked
< 
* transfer closed with outstanding read data remaining
* Closing connection
curl: (18) transfer closed with outstanding read data remaining

```
### Ⅲ. Describe what you expected to happen
返回状态码403,body内容`fail`

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. 配置ai-proxy
2. request block 配置：

```
block_bodies:
- "helloworld"
block_bodys:
- "helloworld"
blocked_code: 403
blocked_message: "fail"
case_sensitive: false
```

### Ⅵ. Environment:

- Higress version:  1.4.0 docker-compse版
- OS : ubuntu20.04
- Others:

### PR DESCRIPTION
**Pull request by CH3CHO**
### Ⅰ. Describe what this PR did

Bypass the response body processing in ai-proxy if it is returned internally.

### Ⅱ. Does this pull request fix one issue?

fixes #1145

### CODE REVIEW

File: plugins/wasm-go/extensions/ai-proxy/main.go [hunk 1]
```diff
@@ -111,6 +111,12 @@ func onHttpRequestBody(ctx wrapper.HttpContext, pluginConfig config.PluginConfig
 }
 
 func onHttpResponseHeaders(ctx wrapper.HttpContext, pluginConfig config.PluginConfig, log wrapper.Log) types.Action {
+	if codeDetails, err := proxywasm.GetProperty([]string{"response", "code_details"}); err == nil && string(codeDetails) != "via_upstream" {
```
Comment (johnlanni) [inline plugins/wasm-go/extensions/ai-proxy/main.go:114]: 这段逻辑要不要在wrapper里加个工具函数，其他地方也可能会用到

**Review by johnlanni — CHANGES_REQUESTED** [review]

File: plugins/wasm-go/extensions/ai-proxy/main.go (hunk 1 above)
Comment (CH3CHO) [reply plugins/wasm-go/extensions/ai-proxy/main.go:114]: 可以

Comment (CH3CHO) [reply plugins/wasm-go/extensions/ai-proxy/main.go]: Done

**Review by johnlanni — APPROVED** [review]
LGTM

**Pull request merged**